pkg/injection/codemodule/installer/job/installer.go

```diff
@@ -0,0 +1,121 @@
+package job
+
+import (
+	"context"
+	"os"
+
+	"github.com/Dynatrace/dynatrace-operator/pkg/controllers/csi/metadata"
+	"github.com/Dynatrace/dynatrace-operator/pkg/injection/codemodule/installer"
+	"github.com/Dynatrace/dynatrace-operator/pkg/injection/codemodule/installer/common"
+	"github.com/Dynatrace/dynatrace-operator/pkg/injection/codemodule/installer/symlink"
+	"github.com/Dynatrace/dynatrace-operator/pkg/util/kubeobjects/env"
+	jobutil "github.com/Dynatrace/dynatrace-operator/pkg/util/kubeobjects/job"
+	"github.com/pkg/errors"
+	"github.com/spf13/afero"
+	k8serrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/utils/ptr"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+type Properties struct {
+	Owner        client.Object
+	ApiReader    client.Reader
+	Client       client.Client
+	ImageUri     string
+	PathResolver metadata.PathResolver
+	PullSecrets  []string
+}
+
+func NewInstaller(ctx context.Context, fs afero.Fs, props *Properties) installer.Installer {
+	return &Installer{
+		fs:       fs,
+		props:    props,
+		nodeName: env.GetNodeName(),
+	}
+}
+
+type Installer struct {
+	fs       afero.Fs
+	props    *Properties
+	nodeName string
+}
+
+func (inst *Installer) InstallAgent(ctx context.Context, targetDir string) (bool, error) {
+	log.Info("installing agent via Job", "image", inst.props.ImageUri, "target dir", targetDir)
+
+	err := inst.fs.MkdirAll(inst.props.PathResolver.AgentSharedBinaryDirBase(), common.MkDirFileMode)
+	if err != nil {
+		log.Info("failed to create the base shared agent directory", "err", err)
+
+		return false, errors.WithStack(err)
+	}
+
+	jobName := inst.buildJobName()
+
+	ready, err := inst.isReady(ctx, targetDir, jobName)
+	if err != nil {
+		return false, err
+	}
+
+	if !ready {
+		return false, nil
+	}
+
+	if err := symlink.CreateForCurrentVersionIfNotExists(inst.fs, targetDir); err != nil {
+		_ = inst.fs.RemoveAll(targetDir)
+
+		log.Info("failed to create symlink for agent installation", "err", err)
+
+		return false, errors.WithStack(err)
+	}
+
+	return true, nil
+}
+
+func (inst *Installer) isReady(ctx context.Context, targetDir, jobName string) (bool, error) {
```

Comment:
I'm just curious: jobName is something we can get from `inst.GetJobName()`; why do we want to pass it as an argument?

and also: IMO it's better to rename for something like `jobIsReady`, cause it's what it does.

Reply:
`getJobName` does hashing, didn't want to do that multiple times, might make sense

Reply:
agree, yes let's keep it.